### ISSUE TITLE
Remove headers from standard http serializers

### ISSUE THREAD

**Issue by mcollina** [pinojs/pino#265]
I do not think they are _that_ needed, and they take more care to manage than the information they provide. Should we change them?

@yunong those are by default included in Bunyan, and you seemed -1 on breaking anything from that. What do you think?

**Comment by davidmarkclements** [comment]
If there's a significant positive performance impact 
(including where that performance impact takes into account the cost of sensitive header redaction) then I'm +1 

We can always supply an option to turn them back on for compat

**Comment by mcollina** [comment]
I think we should remove them all, they are a) unsafe by default and b) force people to use pino-noir 
c) slow (as it in more data being sent over the wire).

**Comment by jsumners** [comment]
I'm torn on this. I think HTTP processing should be performed in HTTP specific modules, e.g. `pino-http` and `hapi-pino`. But those modules rely on the standard serializers from `pino` being available. And the headers of a request are rather important to understanding a request.

**Comment by mcollina** [comment]
Should we just pre-white list some headers that we want to appear, both in the request and the response? I think that would be a good default behavior. How about we extract them to a separate module, so it can be used in pino-http and hapi-pino?

@yoshuawuyts what do you think?

**Comment by jsumners** [comment]
I thought about a configurable whitelist, but that sounds like a big performance hit to me. At this point, I'm thinking it might be best to create `pino-std-serializers` and just let people add them if they need them (e.g. I never even use them except for what `pino-http` and `hapi-pino` do).

**Comment by mcollina** [comment]
I don't think a configurable whitelist is sensible, just a coded-in whitelist. Or something precooked with pino-noir. 

Most headers are not needed, which headers would you need?

**Comment by jsumners** [comment]
That's the thing, it varies from project to project.

**Comment by davidmarkclements** [comment]
The thing is both `pino-http` and `hapi-pino` provide a custom request serializer *anyway* so the built in one isn't even used

On principal we should probably remove the req/res serializers from core, leaving only the error serializer.

**Comment by yoshuawuyts** [comment]
Yeah, I'm not really using the header parsing things from Pino, so don't mind removing them and making them module-specific if it speeds things up.

**Comment by jsumners** [comment]
I vote they be removed to `pino-http-serializers` or all of them to `pino-std-serializers` and just let people use them as they need.

**Comment by yunong** [comment]
@mcollina We're currently using the restify/bunyan serializers, so this isn't going to affect us. However, I would say for us headers are extremely important when it comes to debugging.

**Comment by mcollina** [comment]
@yunong are you logging all headers, are you white-listing or are you black-listing?

**Comment by novemberborn** [comment]
I'd like to share my recent experience (as of ten minutes ago):

Typically I set up `hapi-pino` with a specific `pino` instance. It's quite tricky to know which serializers to pass where. Not having `req` and `res` there by default may make that easier to understand.

I've also found that when redacting just headers it's easier to copy/paste and adjust the serializers than to use `pino-noir`. Specifically I'm interested in redacting cookie headers. `pino-noir` seems to ends up modifying the actual outgoing headers which then crashes my server (https://github.com/pinojs/pino-noir/issues/1#issuecomment-324291778).
